### DISCUSSION TITLE
Self hosted renovate: advanced filtering for selecting repositories

### DISCUSSION THREAD

**Discussion by amimas** [renovatebot/renovate#10465] [Suggest an Idea]
I'm currently self hosting renovate bot and running it in CI as a scheduled pipeline. The bot is currently run as shown below:

```
yarn renovate --autodiscover-filter "gruop1/*"
yarn renovate --autodiscover-filter "group2/*"
```

All of the above is being run in parallel as separate CI jobs in different machines. I'm planning on adding a new group to the pipeline in the same manner. This new group has a lot more repositories than what is currently processed. In my local machine running in dry-run mode, it took about 4 hours to complete.

It would be great if the `autodiscover-filter` could be more more than just glob-pattern, which is only targeting repositories by name. Or, maybe a new filtering option so that the bot could process more repos in parallel.

Both Github and GitLab allows setting "topics" for each repositories. GitLab's API allows listing projects by topics and I assume Github does that also. We can use this for targeting repositories, for example:

```
yarn renovate --autodiscover-filter "group1/*" --filter-by-topics "backend"
yarn renovate --autodiscover-filter "group1/*" --filter-by-topics "frontend"
```

The main goal here is to be able to speed up the processing/execution time by running the bot in parallel.

Because the `autodiscover-filter` option specifically targets repo name, it's a bit constrained. Another option could be enhancing how `autodiscover-filter` behave. For example: should the pattern matching be done against repo name or some other meta data (i.e. topic).

I can see 2 potential issues with targeting repositories by topics, which I haven't thought through yet.

1. How to target/discover repositories without any topics
2. How to avoid targeting same repo twice

For the 2nd item above, a scenario could be one project has topics `backend, service`, a second project has `frontend, service`, and a third project has `monitoring, service`. Then doing the following would be inefficient. However, I think this more of a user/usage issue.

```
yarn renovate --autodiscover-filter "group1/*" --filter-by-topics "backend"
yarn renovate --autodiscover-filter "group1/*" --filter-by-topics "frontend"
yarn renovate --autodiscover-filter "group1/*" --filter-by-topics "service"
```

**Comment by rarkins** [comment]
Related: #10011 #8763

**Comment by amimas** [comment]
I think this is now doable at least as of version 37. There are additional "autodiscover" config options such as: `autodiscoverNamespaces`, `autodiscoverTopics`, etc.
